libcontainer/system/linux.go

```diff
@@ -75,3 +78,29 @@ func Setctty() error {
 	}
 	return nil
 }
+
+/*
+ * Detect whether we are currently running in a user namespace.
+ * Copied from github.com/lxc/lxd/shared/util.go
+ */
+func RunningInUserNS() bool {
+	file, err := os.Open("/proc/self/uid_map")
+	if err != nil {
+		return false
+	}
+	defer file.Close()
+
+	buf := bufio.NewReader(file)
+	l, _, err := buf.ReadLine()
+	if err != nil {
+		return false
+	}
+
+	line := string(l)
+	var a, b, c int64
+	fmt.Sscanf(line, "%d %d %d", &a, &b, &c)
```

Comment:
On Fri, Jan 08, 2016 at 11:51:26AM -0800, Mrunal Patel wrote:
> > +func RunningInUserNS() bool {
> > +	file, err := os.Open("/proc/self/uid_map")
> > +	if err != nil {
> > +		return false
> > +	}
> > +	defer file.Close()
> > +
> > +	buf := bufio.NewReader(file)
> > +	l, _, err := buf.ReadLine()
> > +	if err != nil {
> > +		return false
> > +	}
> > +
> > +	line := string(l)
> > +	var a, b, c int64
> > +	fmt.Sscanf(line, "%d %d %d", &a, &b, &c)
> 
> Can we add a comment that this function assumes that uid_map has been written in the new user namespace (which is a fair assumption)?

I don't understand what you mean.  The file is a procfile, served by the
kernel.  If it's not there we assume the kernel doesn't support user
namespaces and return false at the top.

Happy to add a comment but not sure what you are referring to.